cdap-app-fabric/src/main/java/io/cdap/cdap/internal/events/ProgramStatusEventPublisher.java

```diff
@@ -143,7 +162,23 @@ protected void processMessages(StructuredTableContext structuredTableContext,
     this.eventWriters.stream().forEach(eventWriter -> eventWriter.write(programStatusEvents));
   }
 
-  private boolean shouldPublish(ProgramRunStatus programRunStatus) {
-    return programRunStatus == ProgramRunStatus.STARTING;
+  private boolean shouldPublish(ProgramRunStatus programRunStatus, ProgramRunId programRunId) {
+    return PIPELINE_PROGRAM_NAME.equals(programRunId.getProgram())
```

Comment:
The system namespace check should filter all system programs. Is the check for pipeline program name still required?